source/includes/_data.md

```diff
@@ -0,0 +1,326 @@
+# Data
+
+### Connectors
+
+Connectors offer you a convenient API by which to access data stored across multiple storage providers, from Algorithmia's built-in data store to Amazon S3. You can learn more about the connectors Algorithmia supports [in our documentation](/developers/data). The various connector types are identified as follows:
+
+- `data`: Algorithmia's built-in data storage solution.
+- `azureblob`: Microsoft Azure Blob Storage
+- `dropbox`: Dropbox
+- `gs`: Google Cloud Storage
+- `s3`: Amazon S3
+
+With the exception of `data`, connectors may also be identified by a *label*, or no label if a specific connector is the default for its connector type. These options can be configured for a given connector in your [data portal](/data).
+
+For example, if you create an S3 data connector with the label `test`, you would identify it to the API as `s3+test`. However, if a specific connector is the default for its type, you can omit the label and simply use the connector's type instead. Thus, your default S3 connector would be identified to the API as simply `s3`.
+
+### Data URIs
+
+```yaml
+# Identifies a file housed by Algorithmia's internal storage provider (data://),
+# which belongs to the user "demo" within their "example" collection.
+data://demo/example/test.png
+
+# Identifies a file housed in S3 (connected to via an S3 connector labeled "test"),
+# which resides in the "my_bucket" bucket and "example" directory.
+s3+test://my_bucket/example/test.png
+
+# Identifies a file housed in a user's default Azure Blob data connector,
+# which resides under the "example" directory within the relevant blob container.
+azureblob://example/test.png
+```
+
+Algorithmia supports a *data URI* notation for identifying files and directories across connectors. To construct a data URI, take the unique identifer of a given connector (such as `data`, `azureblob`, or `s3+test`) and supply that as the URI's protocol. Then append the path to the file or directory you wish to act on. See examples to the right.
+
+Algorithmia will return data URIs to you identifying new files or directories you may create, and you can use these in any of our client libraries (including our Python client library) to act on those resources.
+
+Paths are contextual to the data connector you are interacting with. For example, paths for the `data` connector must begin first with the username of user who owns the collection you wish to interact with (or `.my` to refer to the calling user), then the collection name, then the file (`data` does not support directories within collections). However, when interacting with an `s3` connector, you must supply the name of the bucket as the first component of the data. Ensure you carefully read [the documentation](/developers/data) for the connector you are working with.
```

Comment:
Typo: "username of user" -> "username of the user".

Reply:
Also, "you must supply the name of the bucket as the first component of the data" -> "you must supply the name of the bucket as the first component of the data URI".